platforms/android/library-compose/src/main/java/io/element/android/wysiwyg/compose/RichTextEditor.kt

```diff
@@ -0,0 +1,95 @@
+package io.element.android.wysiwyg.compose
+
+import android.text.Editable
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.rememberCoroutineScope
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.viewinterop.AndroidView
+import androidx.core.widget.addTextChangedListener
+import io.element.android.wysiwyg.EditorEditText
+import io.element.android.wysiwyg.view.models.InlineFormat
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.emptyFlow
+import kotlinx.coroutines.launch
+import uniffi.wysiwyg_composer.ActionState
+import uniffi.wysiwyg_composer.ComposerAction
+import uniffi.wysiwyg_composer.MenuAction
+
+/**
+ * A composable rich text editor.
+ *
+ * This composable is a wrapper around the [EditorEditText] view.
+ *
+ * @param modifier The modifier for the layout
+ * @param actions Used to send actions to the editor
+ * @param onHtmlChanged Called whenever the editor content changes with the new HTML
+ * @param onMarkdownChanged Called whenever the editor content changes with the new Markdown
+ * @param onActionsChanged Called whenever the editor action states change
+ * @param onSelectionChanged Called whenever the editor selection changes
+ * @param onMenuActionChanged Called whenever the editor menu action changes
+ */
+@Composable
+fun RichTextEditor(
+    modifier: Modifier = Modifier,
+    actions: Flow<EditorAction> = emptyFlow(),
+    onHtmlChanged: (html: String) -> Unit = { },
+    onMarkdownChanged: (markdown: String) -> Unit = { },
+    onActionsChanged: (actions: Map<ComposerAction, ActionState>) -> Unit = { },
+    onSelectionChanged: (start: Int, end: Int) -> Unit = { _, _ -> },
+    onMenuActionChanged: (menuAction: MenuAction) -> Unit = { },
+) {
+    val scope = rememberCoroutineScope()
+
+    var textWatcher by remember { mutableStateOf<((Editable?) -> Unit)>({ }) }
+
+    AndroidView(
+        modifier = modifier.fillMaxWidth(),
```

Comment:
Since this is a library we shouldn't apply any modifier here (unless it's strictly needed) and leave full control of this view's dimensions to the caller.

Reply:
https://github.com/matrix-org/matrix-rich-text-editor/pull/754/commits/7b67bdd146c2c6ea8ca498c5950e5069d4fd7c93